libs/server/Storage/Session/MainStore/MainStoreOps.cs

```diff
@@ -302,6 +302,72 @@ public unsafe GarnetStatus TTL<TContext, TObjectContext>(ref SpanByte key, Store
             return GarnetStatus.NOTFOUND;
         }
 
+        /// <summary>
+        /// Get the absolute Unix timestamp at which the given key will expire.
+        /// </summary>
+        /// <typeparam name="TContext"></typeparam>
+        /// <typeparam name="TObjectContext"></typeparam>
+        /// <param name="key">The key to get the Unix timestamp.</param>
+        /// <param name="storeType">The store to operate on</param>
+        /// <param name="output">Span to allocate the output of the operation</param>
+        /// <param name="context">Basic Context of the store</param>
+        /// <param name="objectContext">Object Context of the store</param>
+        /// <param name="milliseconds">when true the command to execute is PEXPIRETIME.</param>
+        /// <returns>Returns the absolute Unix timestamp (since January 1, 1970) in seconds or milliseconds at which the given key will expire.</returns>
+        public unsafe GarnetStatus EXPIRETIME<TContext, TObjectContext>(ref SpanByte key, StoreType storeType, ref SpanByteAndMemory output, ref TContext context, ref TObjectContext objectContext, bool milliseconds = false)
+            where TContext : ITsavoriteContext<SpanByte, SpanByte, SpanByte, SpanByteAndMemory, long, MainSessionFunctions, MainStoreFunctions, MainStoreAllocator>
+            where TObjectContext : ITsavoriteContext<byte[], IGarnetObject, ObjectInput, GarnetObjectStoreOutput, long, ObjectSessionFunctions, ObjectStoreFunctions, ObjectStoreAllocator>
+        {
+            int inputSize = sizeof(int) + RespInputHeader.Size;
+            byte* pbCmdInput = stackalloc byte[inputSize];
+
+            byte* pcurr = pbCmdInput;
+            *(int*)pcurr = inputSize - sizeof(int);
+            pcurr += sizeof(int);
+            (*(RespInputHeader*)pcurr).cmd = milliseconds ? RespCommand.PEXPIRETIME : RespCommand.EXPIRETIME;
+            (*(RespInputHeader*)pcurr).flags = 0;
+
+            if (storeType == StoreType.Main || storeType == StoreType.All)
+            {
+                var status = context.Read(ref key, ref Unsafe.AsRef<SpanByte>(pbCmdInput), ref output);
+
+                if (status.IsPending)
+                {
+                    StartPendingMetrics();
+                    CompletePendingForSession(ref status, ref output, ref context);
+                    StopPendingMetrics();
+                }
+
+                if (status.Found) return GarnetStatus.OK;
+            }
+
+            if ((storeType == StoreType.Object || storeType == StoreType.All) && !objectStoreBasicContext.IsNull)
+            {
+                var objInput = new ObjectInput
+                {
+                    header = new RespInputHeader
+                    {
+                        cmd = milliseconds ? RespCommand.PEXPIRETIME : RespCommand.EXPIRETIME,
```

Comment:
FYI - You can't set both cmd and type since they're written to the same area of memory, use only type in this case.

Reply:
Thanks for the info, I didn't notice the `FieldOffset` set to `0` for both the fields, Will have the changes required

Reply:
Fixed it